docs/iot-gateway/integration-with-telegram-bot.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: docwithnav
+title: Integration with Telegram Bot
+
+---
+
+* TOC
+{:toc}
+
+## Overview
+
+Telegram Bots are simply Telegram accounts operated by software – not people – and they'll often have AI features. 
+They can do anything – teach, play, search, broadcast, remind, connect, integrate with other services, or even pass commands to the Internet of Things. 
+You can create your own Telegram Bot and then send him some telemetry data or alarm using Rest API Call Plugin.
+
+
+## Example
+
+In this example, we are going to demonstrate how you can create a Telegram Bot and configure Rest API Call extension to be able to send telemetry messages to Telegram App.
+
+
+### Creation of the Telegram Bot
+
+[The BotFather](https://telegram.me/botfather) is the main bot that will help you to [create](https://core.telegram.org/bots#6-botfather) new bots and change their settings.
+
+Once the creation of the bot is finished, you can generate an authorization token for your new bot. 
+The token is a string along the lines of **'110201543:AAHdqTcvCH1vGWJxfSeofSAs0K5PALDsaw'** that is required to authorize the bot. After that you can start to configure Rest API Call extension.
+
+Prerequisites :
+
+ - ThingsBoard is up and running
+ - Telegram Bot is created
+
+### REST API Call Plugin Integration with Telegram Bot
+
+Let's configure REST API Call plugin first. Go to *Plugins* menu, click on a '+' button and create new plugin:
+
+![image](/images/gateway/telegram-bot/rest-api-telegram-plugin-config.png)
+
+![image](/images/gateway/telegram-bot/rest-api-telegram-plugin-config-2.png)
+
+Please set URI scheme name, URL, port, path, authentication method and headers correctly for the Telegram endpoint where the request is going to be transferred. 
+Base path includes **"/bot" + your Telegram Bot API token**.
+
+Save plugin and click on *'Activate'* plugin button:
+
+![image](/images/gateway/telegram-bot/rest-api-telegram-plugin-config-3.png)
+
+### REST API Call Rule Integration with Telegram Bot
+
+Now it's time to create appropriate Rule.
+
+![image](/images/gateway/telegram-bot/rest-api-telegram-rule-config.png)
+
+Add filter for **POST_TELEMETRY** message type:
+
+![image](/images/reference/plugins/rest-api-call/post-telemetry-filter.png)
+
+Click *'Add'* button to add the filter.
+
+Then select *'REST API Telegram Plugin'* in the drop-down box for the Plugin field:
+
+![image](/images/gateway/telegram-bot/rest-api-telegram-rule-config-2.png)
+
+Add an action that will send temperature telemetry of the device to the Telegram App.
```

Comment:
Please attach a JSON with the rule template. You can leave the keys empty or smth like "PUT YOUR BOT API KEY HERE". It is better to send the message when new alarm is created. This will allow to de-duplicate messages that belong to a same malfunction.